### PR TITLE
fix: remove posts from private sources in most upvoted feed

### PR DESCRIPTION
**Pull request by omBratteng**
Creates a new feed filter to filter out squad posts from `Most upvoted` and `Best discussed` feeds

WT-1550

### CODE REVIEW

**Review by idoshamun — APPROVED** [review]
Looks great to me! Let's wait for @rebelchris

File: __tests__/feeds.ts [hunk 1]
```diff
@@ -1183,6 +1183,35 @@ describe('query mostUpvotedFeed', () => {
...
+    const sourceRepo = con.getRepository(Source);
+    await sourceRepo.update({ id: 'squad' }, { type: SourceType.Squad });
+    const repo = con.getRepository(Post);
+    await repo.update({ id: 'p1' }, { upvotes: 20 });
+    await repo.update({ id: 'p3' }, { upvotes: 15 });
+    await repo.update({ id: 'p2' }, { upvotes: 30, sourceId: 'squad' });
+
+    const res = await client.query(QUERY(30));
+    expect(res.errors).toBeFalsy();
+    expect(res.data).toMatchSnapshot();
```
Comment (rebelchris) [inline __tests__/feeds.ts:1197]: Nothing blocking, but in general snapshots can give us some flakiness on array comparison (as they are not guaranteed order apparently)
For now no action needed, but maybe for future good note to adjust to other evaluations.

File: src/common/feedGenerator.ts [hunk 2]
```diff
@@ -222,6 +222,7 @@ export const applyFeedWhere = (
   removeHiddenPosts = true,
   removeBannedPosts = true,
   allowPrivateSources = true,
+  allowSquadPosts = true,
```
Comment (rebelchris) [inline src/common/feedGenerator.ts:225]: More of a question, is there any existing flow that uses this as default currently?

**Review by rebelchris — APPROVED** [review]
Nothing major, looks good to me.

File: __tests__/feeds.ts (hunk 1 above)
Comment (omBratteng) [reply __tests__/feeds.ts:1197]: Yeah, I noticed that the array order was different between when I ran the all the test and the specific test file

Comment (rebelchris) [reply __tests__/feeds.ts:1197]: Yep, array order of snapshot is not guaranteed unfurtionally (you'll find many github issues in Jest repo for this)
https://dailydotdev.slack.com/archives/C02E2C3C13R/p1690297213537409?thread_ts=1690296315.741339&cid=C02E2C3C13R

File: src/common/feedGenerator.ts (hunk 2 above)
Comment (idoshamun) [reply src/common/feedGenerator.ts:225]: feed and bookmarks

File: __tests__/feeds.ts (hunk 1 above)
Comment (omBratteng) [reply __tests__/feeds.ts:1197]: That's quite unfortunate, since snapshot is a great tool.

Comment (rebelchris) [reply __tests__/feeds.ts:1197]: Agreed! 
Hope they'll either support it soon or we can find a good workaround or something.

As mentioned for now don't worry, let's see if they get caught on flakiness at all.

**Pull request merged**